sympy/solvers/ode.py

```diff
@@ -1052,6 +1045,28 @@ class in it.  Note that a hint may do this anyway if
             else:
                 raise ValueError("Enter boundary conditions of the form ics={f(point}: value, f(x).diff(x, order).subs(x, point): value}")
 
+
+    # Precondition to try remove unnecessary factor and power from equation
+    if isinstance(eq, Pow):
+        eqs_args = eq.args # if f(x)**p=0 then f(x)=0 (p>0)
+        if S(eqs_args[1]).is_positive>0:
+            eq = eqs_args[0]
+
+    # I will try not to change the structure of equation may be user wants to solve the ode by some particular classifier
+    from sympy.polys.polytools import factor
+    eqs = factor(eq)
+    eqs = fraction(eqs)[0] # p/q =0, So we need to solve only p=0
+    eqns = []
+    if isinstance(eqs, Mul):
+        fac = eqs.args
+        for i in fac:
+            if i.has(func):
+                eqns.append(i)
+        if len(eqns)>0:
+            r = {'eqns' : eqns}
+            matching_hints["factorable"] = r
```

Comment:
I would prefer for this to be in a function called e.g. `_ode_factorable_match`. Eventually I want to move all of the solver-specific matching code out of this function.

Reply:
okay